File: gui_view/src/main/scala/com/goexp/galgame/gui/util/res/gameimg/SimpleImage.scala
package com.goexp.galgame.gui.util.res.gameimg

import com.goexp.galgame.common.website.getchu.{GetchuGameLocal, GetchuGameRemote}
import com.goexp.galgame.gui.model.Game

class SimpleImage(private[this] val game: Game) {

  def small(index: Int, src: String) = {
    val remote = GetchuGameRemote.smallSimpleImg(src)
    val local = GetchuGameLocal.smallSimpleImg(game.id, index)
    GameImages.get(game)(local, remote)
  }

  def large(index: Int, src: String) = {
    val remote = GetchuGameRemote.largeSimpleImg(src)
    val local = GetchuGameLocal.largeSimpleImg(game.id, index)
    GameImages.get(game)(local, remote)
  }
}

File: gui_view/src/main/scala/com/goexp/galgame/gui/util/res/gameimg/SimpleImage.java
package com.goexp.galgame.gui.util.res.gameimg;

import com.goexp.galgame.common.website.getchu.GetchuGame;
import com.goexp.galgame.gui.model.Game;
import javafx.scene.image.Image;

public class SimpleImage {

    public static Image small(Game game, int index, String src) {
        final var url = GetchuGame.smallSimpleImg(src);

        return Util.getImage(game, new CacheKey(game.id + "/simple_s_" + index, url));
    }

    public static Image large(Game game, int index, String src) {
        final var url = GetchuGame.largeSimpleImg(src);

        return Util.getImage(game, new CacheKey(game.id + "/simple_l_" + index, url));
    }
}
